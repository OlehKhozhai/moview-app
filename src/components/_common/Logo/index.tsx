import React from 'react';
import cn from 'classnames';
import { Link } from 'react-router-dom';

import styles from './styles.module.scss';

type LogoProps = { className?: string };

const Logo: React.FC<LogoProps> = ({ className }) => {
  return (
    <Link className={cn(styles.root, className)} to="/">
      <span className={styles.bold}>netflix</span>roulete
    </Link>
  );
};

export default React.memo(Logo);
