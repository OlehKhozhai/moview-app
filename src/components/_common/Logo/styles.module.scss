.root {
  font-size: 20px;
  color: #f65261;
  text-decoration: none;
}

.bold {
  font-weight: 800;
}
