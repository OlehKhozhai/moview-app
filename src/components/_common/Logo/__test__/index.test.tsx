import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import Logo from '../index';

describe('Logo component', () => {
  test('should render Logo component by default', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <Logo />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
