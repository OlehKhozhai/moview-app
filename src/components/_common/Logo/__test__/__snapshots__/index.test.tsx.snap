// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render Logo component by default 1`] = `
<a
  className="root"
  href="/"
  onClick={[Function]}
>
  <span
    className="bold"
  >
    netflix
  </span>
  roulete
</a>
`;
